packages/web/pages/api/router/token-out-given-in.ts

```diff
@@ -0,0 +1,147 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  ConcentratedLiquidityPool,
+  ConcentratedLiquidityPoolRaw,
+  CosmwasmPoolRaw,
+  FetchTickDataProvider,
+  OptimizedRoutes,
+  SplitTokenInQuote,
+  StablePool,
+  StablePoolRaw,
+  TokenOutGivenInRouter,
+  TransmuterPool,
+  WeightedPool,
+  WeightedPoolRaw,
+} from "@osmosis-labs/pools";
+
+import { ChainInfos } from "~/config";
+import { queryPaginatedPools } from "~/queries/complex/pools";
+import { queryNumPools } from "~/queries/osmosis";
+
+type Response = {
+  amount: string;
+  split: {
+    initialAmount: string;
+    pools: {
+      id: string;
+    }[];
+    tokenOutDenoms: string[];
+    tokenInDenom: string;
+  }[];
+};
+
+export default async function routeTokenOutGivenIn(req: Request) {
```

Comment:
Do you mind also exposing the candidate route selection logic?

That is, for the given two denoms, return the candidate routes

Reply:
You want candidate routes returned in API?

Reply:
Yeah